backend/src/main/kotlin/no/uib/echo/routes/RegistrationRoutes.kt

```diff
@@ -466,3 +468,31 @@ fun Route.postRegistrationCount() {
         }
     }
 }
+
+fun Route.getUserRegistrations() {
+    get("/user/registrations/{email?}") {
+
+        val email = call.principal<JWTPrincipal>()?.payload?.getClaim("email")?.asString()?.lowercase()
+
+        val userEmail = withContext(Dispatchers.IO) {
+            URLDecoder.decode(call.parameters["email"], "UTF-8")
+        }
+        if (email == null || userEmail == null) {
+            call.respond(HttpStatusCode.BadRequest, "NO EMAIL")
+            return@get
+        }
+
+        if (email != userEmail) {
+            call.respond(HttpStatusCode.Unauthorized)
+            return@get
+        }
+
+        val userRegistrations = transaction {
+            Registration.join(Happening, joinType = JoinType.LEFT, additionalConstraint = { Registration.happeningSlug eq Happening.slug })
+                .select(Registration.userEmail eq email).toList()
+        }.map {
+            it[Registration.happeningSlug]
+        }
```

Comment:
```suggestion
        val userRegistrations = transaction {
            Registration.select {
                userEmail eq email
}.map {
            it[Registration.happeningSlug]
        }
```

Skriver på mobil, men kan ikke denne skrives litt lettere?

Reply:
var vel egt jeg som satte dere på feil spor her en gang way back, sry :))